From 6cc7f49d7052b3f5178539112bafcf5985c3c1b7 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 26 Jul 2025 23:56:17 +0200
Subject: iio: st_sensors: Don't report error when the device is not present on
 I2C bus

On Pinephone, this is not an error, because there are two alternate
magentometer configurations.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/iio/common/st_sensors/st_sensors_core.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/iio/common/st_sensors/st_sensors_core.c b/drivers/iio/common/st_sensors/st_sensors_core.c
index dac593be5695..d19869123e67 100644
--- a/drivers/iio/common/st_sensors/st_sensors_core.c
+++ b/drivers/iio/common/st_sensors/st_sensors_core.c
@@ -599,6 +599,13 @@ int st_sensors_verify_id(struct iio_dev *indio_dev)
 		err = regmap_read(sdata->regmap,
 				  sdata->sensor_settings->wai_addr, &wai);
 		if (err < 0) {
+			if (err == -ENXIO) {
+				dev_info(parent,
+					 "%s is not present on the bus\n",
+					 indio_dev->name);
+				return err;
+			}
+
 			return dev_err_probe(parent, err,
 					     "failed to read Who-Am-I register.\n");
 		}
-- 
2.51.0

